### PR TITLE
Test framework migration

### PR DESCRIPTION
**Pull request by AlexanderSher**
Use Roslyn test utilities instead of `DiagnosticAnalyzerRunner`

### CODE REVIEW

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs [hunk 1]
```diff
@@ -2,75 +2,67 @@
...
-            Assert.Equal("AZC0001", diagnostic.Id);
-            Assert.Equal("Namespace 'RandomNamespace' shouldn't contain public types. Use one of the following pre-approved namespace groups: " +
-                         "Azure.AI, Azure.Analytics, Azure.Data, Azure.Iot, Azure.Media, Azure.Messaging, Azure.Security, Azure.Storage, Microsoft.Extensions.Azure", diagnostic.GetMessage());
+}";
 
-            AnalyzerAssert.DiagnosticLocation(testSource.DefaultMarkerLocation, diagnostic.Location);
+            var diagnostic = Verifier.Diagnostic("AZC0001")
+                .WithMessage("Namespace 'RandomNamespace' shouldn't contain public types. Use one of the following pre-approved namespace groups: " +
+                             "Azure.AI, Azure.Analytics, Azure.Data, Azure.Iot, Azure.Media, Azure.Messaging, Azure.Security, Azure.Storage, Microsoft.Extensions.Azure")
+                .WithSpan(2, 11, 2, 26);
```
Comment (pakrym) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs]: This kind of sucks :)

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs [hunk 2]
```diff
@@ -2,75 +2,67 @@
...
         }
 
         [Fact]
-        public async Task AZC0001ProducedOneErrorPerNamspaceDefinition()
+        public async Task AZC0001ProducedOneErrorPerNamespaceDefinition()
         {
-            var testSource = TestSource.Read(@"
-namespace RandomNamespace
+            const string code = @"
+namespace [|RandomNamespace|]
```
Comment (pakrym) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs]: What is `[|` for?

**Comment by pakrym** [comment]
Thank you for doing the work!

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/MethodBodyAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Azure.ClientSdk.Analyzers 
+{
+    internal readonly struct MethodBodyAnalyzer
```
Comment (pakrym) [inline src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/MethodBodyAnalyzer.cs]: This seems to be new?

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs (hunk 1 above)
Comment (AlexanderSher) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs]: > Can we build something on top of roslyns library?

Your next question about `[||]` answers this one :)
In this particular case, I've used explicit `DiagnosticResult` to ensure message. That is the only case in all tests.

File: src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs (hunk 2 above)
Comment (AlexanderSher) [reply src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers.Tests/AZC0001Tests.cs]: This marks the place where diagnostic would be.

**Comment by AlexanderSher** [comment]
> Haven't noticed async analyzer is also here. Reviewing.

I'm closing this PR now and submit a new one without new analyzers.

**Pull request closed without merging**